backend/src/v5/middleware/dataConverter/inputs/users.js

```diff
@@ -120,6 +120,55 @@ const validateAvatarData = async (req, res, next) => {
 	}
 };
 
+Users.validateForgotPasswordData = async (req, res, next) => {
+	const schema = Yup.object().shape({
+		user: Yup.string().required(),
+	}).strict(true).noUnknown()
+		.required();
+
+	try {
+		await schema.validate(req.body);
+
+		try {
+			const usernameOrEmail = req.body.user;
+			const { user } = await getUserByQuery({ $or: [{ user: usernameOrEmail }, { 'customData.email': usernameOrEmail }] });
```

Comment:
Found a bug here.

Email search needs to be case insensitive. if i add random caps in my email it doesn't find me. 

Can we move this query into model/users.js and make sure it's case insensitive. We're going to run into the same problem when we do registration so let's make sure it's consistent.

Also make sure we add a e2e test case for it

Reply:
PS: we should add projection here, we only watn the username, shouldn't drag the whole user document out

Reply:
we already have getUserByUsername should I just convert that to getUserByUsernameOrEmail ?
![image](https://user-images.githubusercontent.com/71635617/159004769-5b7c8a08-45a4-44dc-8180-0057e6f22f0f.png)

Reply:
nah we want to make sure we don't check for email if the intention is just the username

so probably a new `getUserByUsernameOrEmail`